Libraries/libbom/Tools/lsbom.cpp

```diff
@@ -449,25 +524,15 @@ main(int argc, char **argv)
         /*
          * Print out requested details.
          */
+        std::string toPrint;
         if (options->onlyPath()) {
-            printf("%s\n", path.c_str());
+            toPrint = PrintItemToString(Options::PrintItem::FileName, path, path_info_2_value);
+        } else if (options->printFormat()) {
+            toPrint = PrintItemsToString(*options->printFormat(), path, path_info_2_value);
         } else {
-            // TODO: Respect options about what to print.
-            std::stringstream stream;
-            stream << std::oct << ntohs(path_info_2_value->mode);
-            std::string mode = stream.str();
-            std::string UID = std::to_string(ntohl(path_info_2_value->user));
-            std::string GID = std::to_string(ntohl(path_info_2_value->group));
-            printf("%s\t%s\t%s/%s", path.c_str(), mode.c_str(), UID.c_str(), GID.c_str());
-            
-            if (path_info_2_value->type == bom_path_type_file) {
-                std::string size = std::to_string(ntohl(path_info_2_value->size));
-                std::string checksum = std::to_string(ntohl(path_info_2_value->checksum));
-                printf("\t%s\t%s", size.c_str(), checksum.c_str());
-            }
-            
-            printf("\n");
+            toPrint = PrintItemsToString(GetDefaultPrintFormat(), path, path_info_2_value);
```

Comment:
I notice that in your value_or solution, if printFormat has been filled in via the -p flag, then onlyPath no longer takes precedence over the -p flag as lsbom.
Perhaps you meant:
`std::vector<Options::PrintItem> printFormat = options->onlyPath() ? std::vector<Options::PrintItem>({Options::PrintItem::FileName}) : options->printFormat().value_or(GetDefaultPrintFormat());`

And this too looks good!
I'll get this in there, and thanks again for the feedback!